### ISSUE TITLE
Skinned UI does not work on Wayland.

### ISSUE THREAD

**Issue by tidux** [flathub/org.atheme.audacious#7]
I'm not sure if the skinned UI is doing dirty tricks with multiple X11 windows or what but it's completely non-functional on Wayland sessions.

**Comment by danieldg** [comment]
This might depend on your display manager. It works OK for me using sway, except for being able to drag windows using the skinned titlebar (the window manager's super+drag works fine).

**Comment by tidux** [comment]
I think sway might be using XWayland for that.  I encountered the problem on GNOME.

**Comment by danieldg** [comment]
No, I have disabled access to the x11 socket (and verified it is using wayland).

**Comment by ludg1e** [comment]
I can confirm I can't use skins under Wayland.

OS: Fedora 40
DE: Plasma 6.1 with KWin 6.1 under Wayland

This must be something from the flatpak version because the one from Fedora repos doesn't have this issue.
Both versions don't run under XWayland.

Log:
```
user@fedora:~$ flatpak run org.atheme.audacious
WARNING ../src/libaudqt/audqt.cc:135 [init]: X11/XWayland was not detected. This is unsupported, please do not report bugs.
Detected locale "C" with character encoding "ANSI_X3.4-1968", which is not UTF-8.
Qt depends on a UTF-8 locale, and has switched to "C.UTF-8" instead.
If this causes problems, reconfigure your locale. See the locale(1) manual
for more information.
ERROR ../src/skins-qt/plugin.cc:182 [init]: The Winamp interface is not supported on Wayland. Please run Audacious via XWayland.
WARNING ../src/libaudcore/plugin-init.cc:110 [start_plugin]: Winamp Classic Interface failed to start.
```

Yet the repos version can use skins under Wayland with no issue...
![image](https://github.com/user-attachments/assets/4842fb57-2fe7-46d9-82a0-18e5ed28b223)

**Comment by tidux** [comment]
> This must be something from the flatpak version because the one from Fedora repos doesn't have this issue.

The RPM is silently falling back to X11.  The Flatpak is forcing Wayland.  Check your Flatpak application permissions in KDE's Settings.

**Comment by ludg1e** [comment]
> > This must be something from the flatpak version because the one from Fedora repos doesn't have this issue.
> 
> The RPM is silently falling back to X11. The Flatpak is forcing Wayland. Check your Flatpak application permissions in KDE's Settings.

If the RPM was failing back to X11 then xlsclients would output something, but it doesn't.

**Comment by ludg1e** [comment]
Sorry, you're right. Forced Wayland on the RPM version with QT_QPA_PLATFORM=wayland and the Winamp UI doesn't work.
Though I wonder why doesn't xlsclients output audacious if it's running under xwayland.

My deepest apologies.

**Comment by josh-endries** [comment]
I just ran into this (I think) on Bazzite. Really weird UI behavior when I tried to use Winamp interface. I had to look into the log to find the warnings about Warland, which is how I got here.

Using Flatseal, if I turn off Wayland, regardless of the x11/fallback-x11 toggles, audacious doesn't start up at all (or at least doesn't show a window), so I guess it isn't or can't use XWayland this way... Bummer, I guess I will try QMMP before going full WINE for my llama fix.

**Comment by tidux** [comment]
QMMP also has Wayland skin issues lol.

**Comment by josh-endries** [comment]
Yep, discovered that, and winamp doesn't work well through wine for me. Oh well.
